torch_xla/distributed/xla_dist.py

```diff
@@ -349,8 +395,15 @@ def run(self, cmd):
               'UNHEALTHY_MAINTENANCE')) != 0:
             # TPU Maintenance: kill all training, wait for healthy, and restart
             break
+          if not self._error_queue.empty():
+            # Potential HostError on GCE VM: kill all, wait, and restart
+            self.logger.warning(
+              self._error_queue.get(), extra={'clientip': '', 'ordinal': ''})
```

Comment:
is it possible to have > 1 errors in the error queue in one try?

Reply:
There can be, just that the first error will be the one will be the real reason for the actual error caused. For example, if one host dies the heartbeats may start being messed up but the host dying would be the real reason we should report.

Reply:
sounds like we should empty the queue b4 restarting then, to avoid unnecessary restarts when try n+1 succeeds.

Reply:
I already do that in line 343 at cleanup time to make sure that doesn't happen.

Reply:
looks to me like line 343 is inside`self._cleanup` and it is not called before restarts?

Reply:
ah my bad it is. got it. Thanks!

Reply:
It's called before restarts on line 408.

Reply:
Ah comments crossed. Thanks 😄

Reply:
I blame this on google format using 2 spaces instead of 4 :(